### PR TITLE
Clean/api handler

### PR DESCRIPTION
**Pull request by lukanus**
# What 🕵️‍♀️
Cleaned api package and superflous header write error massages

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -183,29 +178,41 @@ func (a *API) getHeader(w http.ResponseWriter, r *http.Request) (int, error) {
 			"slot":     slot,
 			"proposer": proposer,
 		}).WithError(err).Debug("failed getHeader")
-		return http.StatusBadRequest, err
+
+		w.WriteHeader(http.StatusBadRequest)
```
Comment (aratz-lasa) [inline pkg/api/api.go:182]: can we extract this code into a function? In order to avoid code duplication

File: pkg/api/api.go [hunk 2]
```diff
@@ -183,29 +178,41 @@ func (a *API) getHeader(w http.ResponseWriter, r *http.Request) (int, error) {
...
+		})
+		return
 	}
 
 	m.Observe(a.m.RelayTiming)
 
 	if err = json.NewEncoder(w).Encode(response); err != nil {
 		a.l.WithError(err).WithField("path", r.URL.Path).Debug("failed to write response")
 		a.m.ApiReqCounter.WithLabelValues("getHeader", "500", "response encode").Inc()
-		return http.StatusInternalServerError, err
```
Comment (aratz-lasa) [inline pkg/api/api.go:194]: Why did you delete this?

File: pkg/api/api.go [hunk 3]
```diff
@@ -183,29 +178,41 @@ func (a *API) getHeader(w http.ResponseWriter, r *http.Request) (int, error) {
...
+		return
 	}
 
 	a.m.ApiReqCounter.WithLabelValues("getHeader", "200", "").Inc()
-	return http.StatusOK, nil
```
Comment (aratz-lasa) [inline pkg/api/api.go:198]: Why did you delete this?

**Review by aratz-lasa — CHANGES_REQUESTED** [review]

File: pkg/api/api.go (hunk 2 above)
Comment (lukanus) [reply pkg/api/api.go:194]: as earlier - those functions are just plain http handlers right now.

File: pkg/api/api.go (hunk 1 above)
Comment (lukanus) [reply pkg/api/api.go:182]: If you like. I'm not sure it's needed in this particular case, but If you prefer - I'm fine.

**Review by aratz-lasa — APPROVED** [review]
